Cookbook/Proposals/ComposableReducers.md

```diff
@@ -0,0 +1,181 @@
+# Composable Reducers using Lenses and Prisms
+
+* Author: Yasuhiro Inami
+* Review Manager: David Rodrigues
+
+## Introduction
+
+This proposal adds functional `Lens` and `Prism` ideas to break down `reducer` / `state` / `event` (`action`) in [ReactiveFeedback](https://github.com/Babylonpartners/ReactiveFeedback) so that we can create a large application from small component compositions.
+
+## Motivation
+
+In Babylon.app, we are using [ReactiveFeedback](https://github.com/Babylonpartners/ReactiveFeedback) to control states, side-effects, and feedback loops to define a particular screen (view controller) behavior, owned by `ViewModel`.
+
+However, this `ViewModel` can easily become too complex as `reducer: State -> Event -> State` grows large.
+
+Unfortunately, splitting into multiple `ViewModel`s is not a clever solution, as managing multiple `ReactiveFeedback`s that interact with each other tends to be hard to control.
```

Comment:
Please add further reasons why, or an example of why it's hard. Ideally, this example should be in alternatives considered.

Reply:
I'd like to see this as well, I'd actually prefer to have a solution for splitting view models (+ renderers and other components) into smaller pieces, just doing that for reducers seems a nice but small step. The good example is `PrescriptionViewModel` where it already have smaller reducers via separate functions, and it can probably be improved with this, but ViewModel will still be too big.

I think it will be nice to demonstrate suggested approach on a real life use case with a PR and `PrescriptionViewModel` seems like a good candidate. It will be more work of course but will be a better demonstration. I'm ready to play with it myself on some free time.

Reply:
Added multiple ViewModel example in 47bde8b .